### PR TITLE
Use symlinks for python and pip docker containers

### PR DESCRIPTION
**Pull request by kszucs**
And simplify the docker image definitions.

### CODE REVIEW

**Comment by fsaintjacques** [comment]
@ursabot build

File: ursabot/docker.py [hunk 1]
```diff
@@ -265,87 +277,55 @@ def conda(*packages, files=tuple()):
...
 for arch in ['amd64', 'arm64v8']:
     # UBUNTU
     for version in ['16.04', '18.04']:
         os = f'ubuntu-{version}'
         base = f'{arch}/ubuntu:{version}'
 
         cpp = DockerImage('cpp', base=base, arch=arch, os=os, steps=[
-            RUN(apt(*ubuntu_pkgs))
+            RUN(apt(*ubuntu_pkgs, 'python3', 'python3-pip')),
+            RUN(symlink({'/usr/bin/python': '/usr/bin/python3',
```
Comment (fsaintjacques) [inline ursabot/docker.py]: Symlinking in `/usr/bin` is a dangerous game, you can easily break the base OS like this, e.g. `yum` used to expect python2. Does it work if you use `/usr/local/bin/python`?

File: ursabot/docker.py [hunk 2]
```diff
@@ -265,87 +277,55 @@ def conda(*packages, files=tuple()):
...
     # ALPINE
     for version in ['3.9']:
         os = f'alpine-{version}'
         base = f'{arch}/alpine:{version}'
 
         cpp = DockerImage('cpp', base=base, arch=arch, os=os, steps=[
-            RUN(apk(*alpine_pkgs)),
-            RUN('python -m ensurepip'),
+            RUN(apk(*alpine_pkgs, 'python3-dev', 'py3-pip')),
+            RUN(symlink({'/usr/bin/python': '/usr/bin/python3',
```
Comment (fsaintjacques) [inline ursabot/docker.py]: ditto

File: ursabot/docker.py (hunk 1 above)
Comment (kszucs) [reply ursabot/docker.py]: Good point!

File: ursabot/docker.py [hunk 3]
```diff
@@ -265,87 +276,58 @@ def conda(*packages, files=tuple()):
...
+        python = DockerImage('python-3', base=cpp, steps=python_steps)
+        arrow_images.extend([cpp, python])
 
-        cpp_worker = DockerImage('cpp', base=cpp, tag='worker',
-                                 steps=worker_steps)
-        python_worker = DockerImage('python', base=python, tag='worker',
-                                    steps=worker_steps)
-
-        arrow_images.extend([cpp, python, cpp_worker, python_worker])
+        if version in {'18.04'}:
```
[the remark this answers is not in the document]
Comment (kszucs) [reply ursabot/docker.py]: If We can run the benchmarks on arm64 then We should have an image for it as well. It doesn't mean We'll have builders for it.

File: ursabot/docker.py (hunk 1 above)
Comment (kszucs) [reply ursabot/docker.py]: @fsaintjacques fixed

**Pull request merged**